### PR TITLE
feat(j-s): Service certificate pdf

### PR DESCRIPTION
**Pull request by oddsson**
# Service certificate pdf

[Asana](https://app.asana.com/0/1199153462262248/1208331028610840/f)

## What

Implement a service certificate pdf. 

## Why

This document is needed to be able to tell that a service was done correctly after the case has been archived from RVG

## Screenshots / Gifs

Screenshot 2024-10-17 at 13 42 50
Screenshot 2024-10-17 at 13 44 47
Screenshot 2024-10-17 at 13 46 49

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

## Release Notes

- **New Features**
  - Introduced a new endpoint to retrieve service certificate PDFs for defendants.
  - Added functionality for generating service certificate PDFs with enhanced formatting options.
  - New button added for downloading service certificates in the user interface.

- **Bug Fixes**
  - Corrected terminology in localized strings for better clarity.

- **Enhancements**
  - Improved error handling for missing parameters in service certificate requests.
  - Expanded PDF generation capabilities to include service certificates.
  - Enhanced conditional rendering in the UI for managing service certificates.
  - Updated user role access checks for improved security.

### CODE REVIEW

**Review by gudjong — CHANGES_REQUESTED** [review]
Looks good, except defenders will see a button to open the service certificate, but will not be able to open it. Do give them access, code needs to be added to limited access controllers in the api and backend.

File: apps/judicial-system/backend/src/app/modules/subpoena/limitedAccessSubpoena.controller.ts [hunk 1]
```diff
@@ -82,4 +83,38 @@ export class LimitedAccessSubpoenaController {
...
+    @CurrentCase() theCase: Case,
+    @CurrentDefendant() defendant: Defendant,
+    @CurrentSubpoena() subpoena: Subpoena,
+    @Res() res: Response,
+  ): Promise<void> {
+    this.logger.debug(
+      `Getting service certificate for defendant ${defendantId} in subpoena ${subpoenaId} of case ${caseId} as a pdf document`,
+    )
+
+    if (!subpoena) {
```
Comment (gudjong) [inline apps/judicial-system/backend/src/app/modules/subpoena/limitedAccessSubpoena.controller.ts:108]: Note that there is an alternative subpoena guard that takes care of this. Please consider moving the `SubpoenaExistsOptionalGuard` to the `getSubpoenaPdf` endpoint and use `SubpoenaExistsGuard` for this endpoint. That will allow you to remove this check.

**Review by gudjong — APPROVED** [review]
Looks good. One suggestion.

**Review by gudjong — APPROVED** [review]
Nice. Some failing tests.

File: apps/judicial-system/backend/src/app/modules/subpoena/test/limitedAccessSubpoenaController/limitedAccessSubpoenaControllerGuards.spec.ts [hunk 2]
```diff
@@ -15,7 +14,7 @@ describe('LimitedAccessSubpoenaController - guards', () => {
   })
 
   it('should have the right guard configuration', () => {
-    expect(guards).toHaveLength(7)
+    expect(guards).toHaveLength(6)
```
[the remark this answers is not in the document]
Comment (oddsson) [reply apps/judicial-system/backend/src/app/modules/subpoena/test/limitedAccessSubpoenaController/limitedAccessSubpoenaControllerGuards.spec.ts:17]: Yes it still used in this controller but not on the controller itself but rather an endpoint within the controller, which this test is not testing

**Pull request merged**